packages/core/versioning/core-versioned-http-server-internal/src/internal_versioned_router.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IRouter } from '@kbn/core-http-server';
+import type { VersionedRouter } from '@kbn/core-versioned-http-server';
+import type {
+  VersionedRoute,
+  VersionedRouteConfig,
+  VersionedRouterRoute,
+} from '@kbn/core-versioned-http-server';
+import { InternalVersionedRoute } from './internal_versioned_route';
+import { Method } from './types';
+
+export class InternalVersionedRouter implements VersionedRouter {
```

Comment:
`InternalVersionedRouter`: happy to change this name, this just seemed the most honest 😅

Reply:
Naming is hard, and objectively we're pretty bad a it 😅. 

I'd say that naming that thing `CoreVersionedRouter` would be *slightly* consistent given we already have `CoreKibanaRequest`, but we're also prefixing other things with `Internal`, so... no strong opinion here.